[MinionWolf] ip

MinionWolf opened the pull request.

MinionWolf committed:
changed file name and logo

MinionWolf committed:
echo and list

can echo and list whatever that is added

MinionWolf committed:
mark/unmark

implemented the mark and unmark feature and added a new Task class with constructor and getter and setter

HarshitSrivastavaHS commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,77 @@
+            getList(Arrays.copyOf(list, listCount));
+        }else{
+            addList(reply);
+        }
```
this might have problems, i'm not sure if we've to consider edge cases, however, if someone types:
"learn hyper text markup language", this is supposed to be added to the list (since it doesn't start with any of the given commands), it would still run the first if block and try to mark it as done.

HarshitSrivastavaHS commented:
Overall, looks good except that 1 issue :))

MinionWolf committed:
cleaned up and added classes

ToDo, Deadline, Event classes added

MinionWolf committed:
updated

tried to do inheritance and polymorphism but can't really figure out how to make it work

mendax1234 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,81 @@
+public class Task {
+    private String taskName;
+    private boolean taskStatus;
+    private static int listCount = 0;
+    private static Task[] list = new Task[100];
```
Should the whole task list be managed in the Task class? Or it can be mangaed elsewhere?

mendax1234 replied:
What is the magic number `100` here used for?

Emannuel-Tan replied:
Should the array of Tasks be saved in the Task class or your main class?

mendax1234 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,23 @@
+public class ToDo extends Task{
+
+    private static boolean toDoStatus;
```
Why is the todo Status declared to be `static` here? Does it mean all the todos share one status?

mendax1234 commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.Scanner;
+
+public class Toothless {
```
The class name should be a noun?

mendax1234 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,23 @@
+public class ToDo extends Task{
```
Good job for defining a subclass `Todo` inherited from `Task`.

Emannuel-Tan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,81 @@
+
+public class Task {
+    private String taskName;
+    private boolean taskStatus;
```
Consider changing the variable name of a boolean variable to use the format: <isX>, <wasX>, <hasX>

Emannuel-Tan commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,48 @@
+    public static void main(String[] args) {
+        String logo = "--------------------------------\n" +
+                "Hello I'm Toothless \n" + "What can I do for you? \n" +
+                "--------------------------------";
```
Consider extracting the start and end messages to a function

Emannuel-Tan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,81 @@
+                    "[" + list[listIndex-1].getTaskStatus() + "] " + list[listIndex-1].getTaskName() + "\n" +
+                    "-------------------------------- \n");
+        }
+    }
```
Consider extracting this into 2 seperate functions

Emannuel-Tan commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,48 @@
+        }else if(!mark.isEmpty()){
+            Task.addList(reply,"add");
+        }else{
+            System.out.println("You have entered a wrong entry or command \n");
```
Good job on extracting each command into methods. Maybe extract wrong command output into a seperated method too?

Emannuel-Tan reviewed (commented):
Overall ok, consider extracting more methods and magic numbers

MinionWolf committed:
ToDo done

completed todo

MinionWolf committed:
updated

done with deadline, added event

MinionWolf committed:
completed level 4

MinionWolf committed:
updated

MinionWolf committed:
- completed with level 5
- added new methods to check exception
- added new java exceptions

MinionWolf committed:
deleted unecessary import

MinionWolf committed:
edited description and testing branching

MinionWolf committed:
organized into packages

MinionWolf committed:
Add the text testing and new exceptions

MinionWolf committed:
Merged

MinionWolf committed:
Modified the code to include ArrayList

MinionWolf committed:
Remove unecessary codes

MinionWolf committed:
Add delete function
Update text ui testing with delete function

MinionWolf committed:
Amend the functions to be private

MinionWolf committed:
Add functions to create file, write to file, append to file and fill file

MinionWolf committed:
Add fillFile function
Add the save text file

MinionWolf committed:
Merge branch 'branch-Level-6'

MinionWolf committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Toothless.java

MinionWolf committed:
Update code

MinionWolf committed:
Update list command to show by, from and to for deadline and event

MinionWolf committed:
Implement jar file

MinionWolf committed:
Move main function to the bottom

MinionWolf committed:
Implement writing to file and retrieving content from file

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+import command.Events;
+import command.Task;
+import command.ToDo;
+import exceptions.*;
```
Don't import using wildcard imports

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+        String taskName;
+        String[] tempArray = reply.split(" ");
+        String operation = tempArray[0];
+
```
can be extracted out for better slap

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+            checkOutOfBounds(listIndex);
+            deleteTask(listIndex);
+        }
+        else if(operation.equals("list")){
```
coding standard violation

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+        }
+        else if(operation.equals("list")){
+            getList();
+        }else if(operation.equals("todo")){
```
spacing issues

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+                    "--------------------------------" ;
+
+    //checking the CLI command
+    private static void checkOperation(String reply){
```
method name says checkOperation, but inside, you are also doing the operations. 
separate them out to two methods/classes as necessary (parsing vs execution)

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+        }
+    }
+
+    //add task to list
```
use javadoc comments instead

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+        ToDo td = new ToDo(taskName);
+        tasks.add(td);
+        printTopMessage("todo");
+        System.out.println("[" + td.getType() + "]" + "[" + td.getMarkStatus() + "] " + td.getTaskName() + "\n" +
```
see if you can segregate out the prints to a separate class

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+            count++;
+        }
+
+        printBorder();
```
inconsistent vertical spacing in this class. please standardize for better readability

okkhoy commented on src/main/java/Toothless.java:
```diff
@@ -0,0 +1,368 @@
+
+        System.out.println(bye);
+    }
+}
```
Your main class is doing a lot of heavy lifting. improve the code structure by extracting additional classes.

okkhoy commented on src/main/java/command/Task.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public void setTaskName(String taskName) {
+        this.taskName = taskName;
+    }
+
+    public String getMarkStatus(){
+        return (markStatus ? "X" : " ");
+    }
+
+    public void setMarkStatus(boolean markStatus) {
+        this.markStatus = markStatus;
+    }
+
+    public String getTask(){
+        return "[" + getMarkStatus() + "] " +  getTaskName();
+    }
+
+    public abstract String getType();
+
+    public abstract String getBy();
+
+    public abstract String getFrom();
+
+    public abstract String getTo();
+}
```
you are giving the specification here; hence need to have header comments to indicate what these are for. this is to help developers in subclassing better

okkhoy commented on src/main/java/exceptions/Empty.java:
```diff
@@ -0,0 +1,4 @@
+package exceptions;
+
+public class Empty extends RuntimeException {
```
Inheriting from RuntimeException is not good in this scenario. You should use subclasses of Exception class so that you can handle the exceptions in your codebase.
Additionally, better to have individual exception handling (or atleast specific error messages) for each type of exception instead of having empty classes

okkhoy commented on src/main/java/exceptions/IncompleteFormat.java:
```diff
@@ -0,0 +1,4 @@
+package exceptions;
+
+public class IncompleteFormat extends RuntimeException {
+}
```
Exception naming is incorrect. better would be IncompleteCommandFormatException

okkhoy reviewed (commented):
- SLAP in the code needs improvement
- Code needs some extensive reorganization and refactoring for better structure.
- None of the methods have javadoc comments; it is mandatory to have header comments for every non-tirivial method/class
- There are issues with coding standard and naming; please fix them before final submission

MinionWolf committed:
Categorize java class into Ui, Storage, Parser and Tasklist

MinionWolf committed:
Implement find function for users to search

MinionWolf committed:
Merge branch 'branch-Level-9'

MinionWolf committed:
Include javadoc comments to all non-private methods and classes

MinionWolf committed:
Merge branch 'branch-JavaDoc'

# Conflicts:
#	src/main/java/Toothless.java

MinionWolf committed:
Change theme

MinionWolf committed:
Create _config.yml

MinionWolf committed:
Update README.md

MinionWolf committed:
Update _config.yml

MinionWolf committed:
Update _config.yml

MinionWolf committed:
Update README.md

MinionWolf committed:
Update README.md

MinionWolf committed:
Update README.md

MinionWolf committed:
Update README.md

MinionWolf committed:
Update javadoc comment
Clean code

MinionWolf committed:
Update javadoc comment
Clean code

MinionWolf committed:
Clean code

MinionWolf committed:
Clean code

MinionWolf committed:
Add method to format date

MinionWolf committed:
Merge branch 'branch-Level-8'

MinionWolf committed:
Update README.md and Ui.Java

MinionWolf committed:
Update README.md

MinionWolf committed:
Change config.yml to different theme

MinionWolf committed:
Update README.md
